projects/addon-editor/extensions/indent-outdent.ts

```diff
@@ -1,217 +1,19 @@
-        const {name} = node.type;
-
-        if (name === `paragraph` || name === `heading`) {
-            tr = setNodeIndentMarkup(tr, pos, delta);
-
-            return false;
-        }
-
-        return !isListNode(node);
-    });
-
-    return tr;
-}
-
-export const Indent = Extension.create<IndentOptions>({
+// TODO: rename to TuiTabExtension in 3.0
+export const Indent = Extension.create({
     name: `indent`,
 
-    addOptions(): IndentOptions {
-        return {
-            types: [`heading`, `paragraph`],
-            indentLevels: [0, 30, 60, 90, 120, 150, 180, 210],
-            defaultIndentLevel: 0,
-        };
-    },
-
-    addGlobalAttributes(): GlobalAttributes {
-        return [
-            {
-                types: this.options.types,
-                attributes: {
-                    indent: {
-                        default: this.options.defaultIndentLevel,
-                        renderHTML: ({indent}) =>
-                            indent
-                                ? {
-                                      style: `margin-left: ${indent}px;`,
-                                  }
-                                : {},
-                        parseHTML: ({style}) =>
-                            parseInt(style.marginLeft, 10) ||
-                            this.options.defaultIndentLevel,
-                        keepOnSplit: false,
-                    },
-                },
-            },
-        ];
-    },
-
-    addCommands(): {indent: () => Command; outdent: () => Command} {
-        return {
-            indent:
-                () =>
-                ({tr, state, dispatch}) => {
-                    const {selection} = state;
-
-                    tr = tr.setSelection(selection);
-                    tr = updateIndentLevel(tr, IndentProps.more);
-
-                    if (tr.docChanged && dispatch) {
-                        dispatch(tr);
-
-                        return true;
-                    }
-
-                    return false;
-                },
-            outdent:
-                () =>
-                ({tr, state, dispatch}) => {
-                    const {selection} = state;
-
-                    tr = tr.setSelection(selection);
-                    tr = updateIndentLevel(tr, IndentProps.less);
-
-                    if (tr.docChanged && dispatch) {
-                        dispatch(tr);
-
-                        return true;
-                    }
-
-                    return false;
-                },
-        };
-    },
-
     addKeyboardShortcuts(): any {
         return {
             Tab: () =>
                 tuiIsOrderedOrBulletList(this.editor)
                     ? null
-                    : this.editor.commands.indent(),
-            'Shift-Tab': () =>
-                tuiIsOrderedOrBulletList(this.editor)
-                    ? null
-                    : this.editor.commands.indent(),
+                    : this.editor.commands.insertContent(`\t`),
```

Comment:
Does this mean all editor and sockets have to have `white-space: pre-wrap`?

Reply:
Editor already has `white-space: pre-wrap`. And user can add it to socket like in example 4, if he want to display multiple spaces/tab. Or do you think we should add it to socket by default ?

Reply:
`Or do you think we should add it to socket by default ` it's good point I think so